File: common/src/main/java/com/thesilentnights/commands/admin/AdminCommands.java
package com.thesilentnights.commands.admin;

import com.mojang.brigadier.CommandDispatcher;
import com.mojang.brigadier.builder.LiteralArgumentBuilder;
import net.minecraft.commands.CommandSourceStack;
import net.minecraft.commands.Commands;

public interface AdminCommands {
    static void register(CommandDispatcher<CommandSourceStack> dispatcher){
        LiteralArgumentBuilder<CommandSourceStack> mainNode = Commands.literal("easylogin").requires(commandSourceStack -> commandSourceStack.hasPermission(4));
        dispatcher.register(new PlayerInfoCommands().getCommand(mainNode));
        dispatcher.register(new TeleportToOfflinePlayer().getCommand(mainNode));
    }

    LiteralArgumentBuilder<CommandSourceStack> getCommand(LiteralArgumentBuilder<CommandSourceStack> mainNode);
}

File: src/main/kotlin/com/thesilentnights/easylogin/commands/server/admin/AdminCommands.kt
package com.thesilentnights.easylogin.commands.server.admin

import com.mojang.brigadier.builder.LiteralArgumentBuilder
import com.thesilentnights.easylogin.service.LoginService
import net.minecraft.commands.CommandSourceStack
import net.minecraft.commands.Commands
import net.minecraft.world.entity.Entity

interface AdminCommands {
    fun getCommand(mainNode: LiteralArgumentBuilder<CommandSourceStack>): LiteralArgumentBuilder<CommandSourceStack>

    companion object {
        val mainNode: LiteralArgumentBuilder<CommandSourceStack> =
            Commands.literal("easylogin")
                .requires { commandSourceStack: CommandSourceStack ->
                    val entity: Entity? = commandSourceStack.entity
                    if (entity == null) {
                        //execute from terminal
                        return@requires commandSourceStack.hasPermission(4)
                    } else {
                        return@requires commandSourceStack.hasPermission(4) && LoginService.isLoggedIn(entity.getUUID())
                    }
                }
    }
}

